### PR TITLE
Zijian/more refactoring

### PR DESCRIPTION
**Pull request by BobbyZhouZijian**

### CODE REVIEW

File: src/main/java/seedu/address/logic/parser/Parser.java [hunk 1]
```diff
@@ -7,6 +7,8 @@
 import seedu.address.logic.parser.exceptions.ParseException;
 
 
+
```
Comment (MarcTzh) [inline src/main/java/seedu/address/logic/parser/Parser.java]: Remove

File: src/test/java/seedu/address/logic/parser/FindTaskCommandParserTest.java [hunk 2]
```diff
@@ -33,18 +35,19 @@ public void parse_validArgs_returnsFindCommand() {
...
         predicate.setKeyword(PREFIX_TITLE, "Meet Alice");
-        predicate.setKeyword(PREFIX_DESCRIPTION, "play");
+        predicate.setKeyword(PREFIX_DESCRIPTION, "project");
         predicate.setKeyword(PREFIX_DATE, "01-01-2020");
         predicate.setKeyword(PREFIX_STATUS, "incomplete");
+        predicate.setKeyword(PREFIX_TAG, "CS2103T");
         FindTaskCommand expectedFindTaskCommand = new FindTaskCommand(predicate);
         assertParseSuccess(parser,
-                " title:Meet Alice desc:play date:01-01-2020 status: incomplete",
+                " title:Meet Alice desc:project date:01-01-2020 status: incomplete tag:CS2103T",
```
Comment (MarcTzh) [inline src/test/java/seedu/address/logic/parser/FindTaskCommandParserTest.java]: I think the space before status is not intended right

File: src/test/java/seedu/address/logic/parser/FindTaskCommandParserTest.java [hunk 3]
```diff
@@ -53,14 +56,15 @@ public void parse_emptyArgs_returnsFalse() {
         // throw error if argument is empty
         assertParseFailure(parser, " title:", MESSAGE_EMPTY_SEARCH_PHRASE);
         assertParseFailure(parser, " desc:", MESSAGE_EMPTY_SEARCH_PHRASE);
-        assertParseFailure(parser, " status:", MESSAGE_EMPTY_SEARCH_PHRASE);
         assertParseFailure(parser, " date:", MESSAGE_EMPTY_SEARCH_PHRASE);
+        assertParseFailure(parser, " tag:", MESSAGE_EMPTY_SEARCH_PHRASE);
+        assertParseFailure(parser, " status:", MESSAGE_EMPTY_SEARCH_PHRASE);
```
Comment (MarcTzh) [inline src/test/java/seedu/address/logic/parser/FindTaskCommandParserTest.java:61]: Hmm what about time or datetime

File: src/test/java/seedu/address/logic/parser/FindTaskCommandParserTest.java [hunk 4]
```diff
@@ -71,6 +75,7 @@ public void parse_argsWrongFormat_returnsFalse() {
         assertParseFailure(parser, " status:comple", Status.SEARCH_CONSTRAINTS);
         assertParseFailure(parser, " date:01-01-202", DateTimeUtil.SEARCH_DATE_CONSTRAINTS);
         assertParseFailure(parser, " date:13", DateTimeUtil.SEARCH_DATE_CONSTRAINTS);
+        assertParseFailure(parser, " tag:##", Tag.SEARCH_CONSTRAINTS);
```
Comment (MarcTzh) [inline src/test/java/seedu/address/logic/parser/FindTaskCommandParserTest.java:79]: Maybe you should include the test for time and desc as well? Test the regex

File: src/test/java/seedu/address/logic/parser/PlanusParserTest.java [hunk 5]
```diff
@@ -71,18 +84,36 @@ public void parseCommand_find() throws Exception {
...
 
+    @Test
+    public void parseCommand_findLesson() throws Exception {
+        List<String> keywords = Arrays.asList("title:tutorial", "tag:cs2103");
+        FindLessonCommand command = (FindLessonCommand) parser.parseCommand(
+                FindLessonCommand.COMMAND_WORD + " " + String.join(" ", keywords));
+
+        LessonContainsKeywordsPredicate predicate = new LessonContainsKeywordsPredicate();
+        predicate.setKeyword(PREFIX_TITLE, "tutorial");
+        predicate.setKeyword(PREFIX_TAG, "cs2103");
```
Comment (MarcTzh) [inline src/test/java/seedu/address/logic/parser/PlanusParserTest.java:95]: Do lower case tags work?

**Review by MarcTzh — COMMENTED** [review]
Some nits

File: src/test/java/seedu/address/logic/parser/FindTaskCommandParserTest.java (hunk 3 above)
Comment (BobbyZhouZijian) [reply src/test/java/seedu/address/logic/parser/FindTaskCommandParserTest.java:61]: seems FindTask current cannot find by time

File: src/test/java/seedu/address/logic/parser/PlanusParserTest.java (hunk 5 above)
Comment (BobbyZhouZijian) [reply src/test/java/seedu/address/logic/parser/PlanusParserTest.java:95]: yes

**Review by MarcTzh — APPROVED** [review]
LGTM

**Pull request merged**